### ISSUE TITLE
Terrain Texture: ORM map not applied properly

### ISSUE THREAD

**Issue by lukaspj** [TorqueGameEngines/Torque3D#422]
There is some bad math in the calculation of the ORM map, only the first layer with ORM is applied properly:
![image](https://user-images.githubusercontent.com/2369614/103152591-ccfa9d00-4789-11eb-8abd-057e8a75f946.png)

```hlsl
// Composite Matinfo map 0
float3 matinfoCol0 = lerp(float3(1.0,1.0,0.0),ormMapArray.Sample(ormMapSampler, float3(IN.detCoord0.xy,detailIdStrengthParallax[0].x)).rgb, detailBlend0);
OUT.col2 = float4(0.0,matinfoCol0);

// Composite Matinfo map 1
float3 matinfoCol1 = ormMapArray.Sample(ormMapSampler, float3(IN.detCoord1.xy, detailIdStrengthParallax[1].x)).rgb*detailBlend1;
OUT.col2.gba += matinfoCol1;
```

> Wouldn't that set it to (1,1,0) after the first composite map, and then when you add the second composite map, those would get above 1 since you add it?

**Comment by lukaspj** [comment]
I tried changing it to:

```hlsl
   // Composite Matinfo map 0
   OUT.col2 = float4(0.0, 1.0, 1.0, 0.0);
   float3 matinfoCol0 = lerp(OUT.col2.gba, ormConfigMapTex0.Sample(ormConfigMap0, IN.detCoord0.xy).rgb, detailBlend0);
   OUT.col2.gba = matinfoCol0;
   
   // Composite Matinfo map 1
   float3 matinfoCol1 = lerp(OUT.col2.gba, ormConfigMapTex1.Sample(ormConfigMap1, IN.detCoord1.xy).rgb, detailBlend1);
   OUT.col2.gba = matinfoCol1;
```

And it seems better:
![image](https://user-images.githubusercontent.com/2369614/103152653-6aee6780-478a-11eb-9bc7-619a3f8cabfa.png)

But there is still several oddities around the ORM map.

For example, inverting the roughness is completely broken:
![image](https://user-images.githubusercontent.com/2369614/103152670-8fe2da80-478a-11eb-94f5-020817e6e228.png)

**Comment by lukaspj** [comment]
Fixed by #424
